hedera-node/src/main/java/com/hedera/services/store/models/Account.java

```diff
@@ -94,6 +95,8 @@ public void associateWith(List<Token> tokens, int maxAllowed) {
 
 	public void dissociateWith(List<Token> tokens) {
 		final Set<Id> uniqueIds = new HashSet<>();
+		validateTrue(ownedNfts == 0, FAIL_INVALID, () ->
```

Comment:
`ownedNfts` is property on the account level that refers to "all tokens". If you want to disassociate fungible token, and you have another nonfungible token, ownedNfts will be !=0 but you must be able to disassociate. 

You have to check the balance of the account for the given token that you are disassociating